infrastructure/code/code_html_test.go

```diff
@@ -25,50 +27,261 @@ import (
 	"github.com/snyk/snyk-ls/internal/testutil"
 )
 
-func Test_CodeDetailsPanel_html_withDataFlow(t *testing.T) {
+func Test_CodeDetailsPanel_html_getDetailsHtml(t *testing.T) {
 	_ = testutil.UnitTest(t)
-	expectedVariables := []string{"${headerEnd}", "${cspSource}", "${nonce}", "${severityIcon}"}
 
+	dataFlow := getDataFlowElements()
+	fixes := getFixes()
+	repoCount := 54387
 	issue := snyk.Issue{
 		ID:       "java/DontUsePrintStackTrace",
-		Severity: 3,
+		Severity: 2,
 		AdditionalData: snyk.CodeIssueData{
-			DataFlow: []snyk.DataFlowElement{
-				{
-					Content:  "if (!vulnLines.every(e => selectedLines.includes(e))) return false",
-					FilePath: "/Users/cata/git/juice-shop/routes/vulnCodeSnippet.ts",
-					FlowRange: snyk.Range{
-						End: snyk.Position{
-							Character: 42,
-							Line:      67,
-						},
-						Start: snyk.Position{
-							Character: 28,
-							Line:      67,
-						},
-					},
-					Position: 0,
-				},
-			},
+			Title:              "Allocation of Resources Without Limits or Throttling",
+			DataFlow:           dataFlow,
+			ExampleCommitFixes: fixes,
+			RepoDatasetSize:    repoCount,
+			IsSecurityType:     true,
+			Message:            "Either rethrow this java.lang.InterruptedException or set the interrupted flag on the current thread with 'Thread.currentThread().interrupt()'. Otherwise the information that the current thread was interrupted will be lost.",
 		},
 	}
 
 	// invoke method under test
-	issueDetailsPanelHtml := getDetailsHtml(issue)
+	codePanelHtml := getDetailsHtml(issue)
+
+	// assert Data Flow section
+	expectedDataFlowHeading := fmt.Sprintf("<h2>Data Flow - %d steps</h2>", len(dataFlow)) // TODO: handle pluralization
```

Comment:
There's a TODO left here - was this addressed or still need to do? Or an improvement on the existing code maybe so we can leave it for later

Reply:
I need to check how we currently handle pluralization in IntelliJ logic and I'll apply it here, thanks!!

Reply:
Done 🙌